src/colab_gitlab/signals.py

```diff
@@ -0,0 +1,105 @@
+
+import logging
+
+import requests
+
+from django.conf import settings
+from django.dispatch import receiver
+
+from colab.accounts.signals import user_password_changed, user_created
+
+LOGGER = logging.getLogger('colab.plugins.gitlab')
+
+
+@receiver(user_password_changed)
+def update_gitlab_password(sender, **kwargs):
+    user = kwargs.get('user')
+    password = kwargs.get('password')
+
+    app_config = settings.COLAB_APPS.get('colab_gitlab', {})
+    private_token = app_config.get('private_token')
+    upstream = app_config.get('upstream', '').rstrip('/')
+
+    error_msg = u'Error trying to update "%s" password on Gitlab. Reason: %s'
+
+    users_endpoint = '{}/api/v3/users'.format(upstream)
+
+    params = {'username': user.username, 'private_token': private_token}
```

Comment:
This username param does not do anything for the users_endpoint... to filter by username or email, use 'search'. eg: 'api/v3/users?search=user_name&private_token=xyz'
See section List Users(for admins) in https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/users.md